readthedocs/proxito/views/serve.py

```diff
@@ -265,6 +265,14 @@ def _get_canonical_redirect_type(self, request):
                 log.debug("Proxito CNAME HTTPS Redirect.", domain=domain.domain)
                 return RedirectType.http_to_https
 
+        if (
+            unresolved_domain.is_from_public_domain
+            or unresolved_domain.is_from_external_domain
+        ):
+            if settings.PUBLIC_DOMAIN_USES_HTTPS and not request.is_secure():
```

Comment:
Having 2 if's nested with 1 piece of logic feels weird. I think we want 1 big `if`, even though it will be complicated?